plugins/catalog-react/src/hooks/useEntityListProvider.tsx

```diff
@@ -232,9 +233,17 @@ export const EntityListProvider = <EntityFilters extends DefaultEntityFilters>(
             compact(Object.values(outputState.appliedFilters)),
           );
 
-          if (!isEqual(previousBackendFilter, backendFilter)) {
+          if (
+            !isEqual(previousBackendFilter, backendFilter) ||
+            requestedFilters.text?.value !== outputState.textSearch
```

Comment:
Hmm, this and the below are related. The existing filter interface is pretty clearly designed to work only for `filters` in the `getEntities`/`queryEntities` API. Unless we add a new method to `EntityFilter` like `getFilterPath: () => string` that works with `queryEntities`' additional parameters, I'm not sure we can do this in a workable backwards compatible way. Ie if we add a `getCatalogFilters` method to `EntityTextSearch` the `fullTextFilter` would get picked up for non-`queryEntities` endpoints.

Reply:
I think your idea of adding a method is a good one as the new method is going to return some data which won't go into the backend filters but somewhere else.
We could also implement `getCatalogFilters` and return an empty object, so that `EntityTextSearch` will be treated as backend filter.

Reply:
Updated in https://github.com/backstage/backstage/pull/22131/commits/03604cffa2f5b8d39879e906ae309056fcb18dd1. This is a hybrid approach that uses the preexisting `getCatalogFilters` for default values and then `getBackendRequestParameters` for more overrideable values. Open to naming suggestions here.